src/Repository.cc

```diff
@@ -594,6 +594,27 @@ bool Repository::branch_up_to_date(const std::string& branch_name)
 }
 #endif
 
+void Repository::checkout(const std::string& branch_name, const std::vector<std::string>& paths)
+{
+     
+    git_checkout_options checkout_opts = GIT_CHECKOUT_OPTIONS_INIT;
+    // define the paths of files to checkout
+    checkout_opts.paths.count = paths.size();
+    checkout_opts.paths.strings = (char **) paths.data();
```

Comment:
C style cast

Reply:
A review was not planned yet, I was just requesting a merge so that Lars can see if that is what he needs. Probably I should have asked him eye-to-eye. But thanks anyway, I will prepare for review tomorrow.

Reply:
> A review was not planned yet, I was just requesting a merge so that Lars can see if that is what he needs. Probably I should have asked him eye-to-eye. But thanks anyway, I will prepare for review tomorrow.

@wyrnat: I have converted this into a draft as long as it is not ready for review. Please mark it as "ready for review" whenever you are ready.

Reply:
> I will prepare for review tomorrow.

Please define `tomorrow` ;-D

Just joking.

Reply:
@Finii tomorrow is today :P